pkg/qsfsd/qsfs.go

```diff
@@ -76,18 +73,20 @@ func setQSFSDefaults(cfg *zos.QuantumSafeFS) zstorConfig {
 func (q *QSFS) Mount(wlID string, cfg zos.QuantumSafeFS) (info pkg.QSFSInfo, err error) {
 	defer func() {
 		if err != nil {
-			if err := q.Unmount(wlID); err != nil {
-				log.Error().Err(err).Msg("error cleaning up after qsfs setup failure")
-			}
+			q.Unmount(wlID)
 		}
 	}()
 	zstorConfig := setQSFSDefaults(&cfg)
 	networkd := stubs.NewNetworkerStub(q.cl)
 	flistd := stubs.NewFlisterStub(q.cl)
 	contd := stubs.NewContainerModuleStub(q.cl)
 
-	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	ctx, cancel := context.WithTimeout(context.Background(), 40*time.Second)
```

Comment:
Why 40 seconds is better then 60 seconds?

Reply:
Some info:
- It takes ~7 seconds to deploy on fresh zdbs.
- If the zdbs have data, the container will try to download it before exposing the mountpoint. This download phase time can't be known here.
- The container only downloads indexes and the a data file of size 16MB have an index of size 11KB. So if the qsfs was mounted before with 1 TB data written to it, it'll download ~700 MB.

The time to finish this depends on the network speed. And during this time provisiond will be blocked. There's no proper error reporting from the container to zos, when it fails, zinit tries to restart it indefinitely. So the full waiting time will be done for EVERY failed configuration.

This is ofc is no reason for why 40 seconds is better than a minute, but it looks like the fix as you mentioned should be in provisiond.